### PR TITLE
Update `toPythonObj` to use new getters -- part 1

### PR DESCRIPTION
**Pull request by yoni206**
Following #6496 , this PR adds new getters to the python API, as well as tests for them. This makes `toPythonObj` simpler.

A future PR will add more getters to the python API.

Note: I ran `yapf` on the test file, so there are quite a few white-space diffs. It is recommended to check this box:
image

Co-authored-by: Gereon Kremer <nafur42@gmail.com>

### CODE REVIEW

File: src/api/python/cvc5.pxi [hunk 1]
```diff
@@ -1603,126 +1645,37 @@ cdef class Term:
...
-          Boolean -- returns a Python bool
-          Int     -- returns a Python int
-          Real    -- returns a Python Fraction
-          BV      -- returns a Python int (treats BV as unsigned)
-          Array   -- returns a Python dict mapping indices to values
-                  -- the constant base is returned as the default value
-          String  -- returns a Python Unicode string
-        '''
+    def getIntegerValue(self):
+        return self.cterm.getIntegerValue().decode()
```
Comment (nafur) [inline src/api/python/cvc5.pxi:1666]:
```suggestion
        return int(self.cterm.getIntegerValue().decode())
```

File: src/api/python/cvc5.pxi [hunk 2]
```diff
@@ -1603,126 +1645,37 @@ cdef class Term:
...
-              i += len(escape_prefix) + len(val) + 1
-        else:
-            raise ValueError("Cannot convert term {}"
-                             " of sort {} to Python object".format(string_repr,
-                                                                   sort))
+    def isRealValue(self):
+        return self.cterm.isRealValue()
+
+    def getRealValue(self):
+        return self.cterm.getRealValue().decode()
```
Comment (nafur) [inline src/api/python/cvc5.pxi:1672]:
```suggestion
        return float(fractions.Fraction(self.cterm.getRealValue().decode()))
```

**Review by nafur — COMMENTED** [review]
What about `isConstArray()`, `isAbstractValue()`, `isTupleValue()`, `isFloatingPoint*()`, `isSetValue()`, `isSequenceValue()`, `isUninterpretedValue()` (and the corresponding `get*()`)?

Also: I still push for using appropriate native types where available, i.e. python numerals instead of strings.

**Comment by yoni206** [comment]
Thanks for the review! Your code suggestions are committed in 84c2b70, and the adjustments to the tests (using python native types and not strings) is committed in a830311.

**Comment by yoni206** [comment]
I have updated the PR according to a discussion on slack, in which it was decided to keep both the getters and the `toPythonObj` function.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**